src/main/java/org/opensearch/flowframework/workflow/CreateConnectorStep.java

```diff
@@ -69,11 +81,41 @@ public CompletableFuture<WorkflowData> execute(List<WorkflowData> data) throws I
 
             @Override
             public void onResponse(MLCreateConnectorResponse mlCreateConnectorResponse) {
-                logger.info("Created connector successfully");
-                // TODO Add the response to Global Context
                 createConnectorFuture.complete(
                     new WorkflowData(Map.ofEntries(Map.entry("connector_id", mlCreateConnectorResponse.getConnectorId())))
                 );
+                try {
+                    logger.info("Created connector successfully");
+                    String workflowId = data.get(0).getWorkflowId();
+                    String workflowStepName = getName();
+                    ResourcesCreated newResource = new ResourcesCreated(workflowStepName, mlCreateConnectorResponse.getConnectorId());
+                    XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+                    newResource.toXContent(builder, ToXContentObject.EMPTY_PARAMS);
+
+                    // The script to append a new object to the resources_created array
+                    Script script = new Script(
+                        ScriptType.INLINE,
+                        "painless",
+                        "ctx._source.resources_created.add(params.newResource)",
+                        Collections.singletonMap("newResource", newResource)
+                    );
+
+                    flowFrameworkIndicesHandler.updateFlowFrameworkSystemIndexDocWithScript(
+                        WORKFLOW_STATE_INDEX,
+                        workflowId,
+                        script,
+                        ActionListener.wrap(updateResponse -> {
+                            logger.info("updated resources craeted of {}", workflowId);
+                        }, exception -> {
+                            createConnectorFuture.completeExceptionally(
+                                new FlowFrameworkException(exception.getMessage(), ExceptionsHelper.status(exception))
+                            );
+                            logger.error("Failed to update workflow state with newly created resource: {}", exception);
+                        })
+                    );
+                } catch (IOException e) {
+                    logger.error("Failed to parse new created resource", e);
+                }
```

Comment:
Let's pull this code block a new private method in ParseUtil class and call that method inside             `createConnectorFuture.complete(`.

We can use the same method to update the resources_created for all the workflow_steps which is missing in this PR.

Reply:
Io am okay trying to abstract this in future PR to resolve this [issue](https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/136), not to block actual get status API content